File: app/src/main/java/mx/edu/upqroo/kristenandroid/viewModels/KardexViewModel.java
package mx.edu.upqroo.kristenandroid.viewModels;

import android.app.Application;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import mx.edu.upqroo.kristenandroid.data.database.entities.Kardex;
import mx.edu.upqroo.kristenandroid.data.repositories.KardexRepository;

public class KardexViewModel extends AndroidViewModel {
    private KardexRepository mRepo;

    public KardexViewModel(@NonNull Application application) {
        super(application);
        mRepo = KardexRepository.getInstance(application);
    }

    public LiveData<List<Kardex>> getKardex(String userId) {
        return mRepo.getKardex(userId);
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/viewModels/KardexViewModel.kt
package mx.edu.upqroo.kristenandroid.viewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import mx.edu.upqroo.kristenandroid.data.database.entities.Kardex
import mx.edu.upqroo.kristenandroid.data.repositories.KardexRepository

class KardexViewModel(application: Application) : AndroidViewModel(application) {
    private val mRepo: KardexRepository

    init {
        mRepo = KardexRepository.getInstance(application)
    }

    fun getKardex(userId: String): LiveData<List<Kardex>> {
        return mRepo.getKardex(userId)
    }

    fun updateKardexFromService() {
        mRepo.updateKardexFromService()
    }
}
